lib/dataset/variable_instantiate_dataset.cpp

```diff
@@ -9,6 +9,23 @@
 
 namespace scipp::variable {
 
+namespace {
+std::string format_data_array(const DataArray &da) {
+  auto s = "DataArray(dims=" + to_string(da.dims()) +
+           ", dtype=" + to_string(da.dtype());
+  if (da.unit() != units::none)
+    s += ", unit=" + to_string(da.unit());
+  return s + ')';
+}
+} // namespace
+
+template <>
+std::string Formatter<DataArray>::format(const Variable &var) const {
+  if (var.dims().volume() == 1)
+    return "DataArray" + format_variable_like(var.value<DataArray>());
```

Comment:
So is this mainly designed for the case where we have a scalar variable that contains a DataArray? I guess this covers 90% of cases?
Will this issue become less frequent if Datagroup is used?

Reply:
I think we are long way from being able to replace the `attrs` property with `DataGroup`, since the former is implemented in C++. However, `DataGroup` opens up ways to store these on a higher level, so in a sense yet.

I have not seen many cases with these dtypes that were not simply a scalar variable wrapping them, so probably more than 90%.